crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -105,6 +106,99 @@ pub(super) const END_EXPR_SET: TokenSet = TokenSet::new([
 /// Tokens that can appear at the end of a sequence.
 const END_SEQUENCE_SET: TokenSet = END_EXPR_SET.remove(TokenKind::Comma);
 
+/// Represents the expression parsing context.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub(super) struct ExpressionContext(ExpressionContextFlags);
+
+impl Default for ExpressionContext {
+    /// Sets the default flags for the expression context which is to:
+    /// * Consider `in` keyword in a comparison expression
+    /// * Disallow starred and yield expressions
+    fn default() -> Self {
+        ExpressionContext(ExpressionContextFlags::INCLUDE_IN)
+    }
+}
+
+bitflags! {
+    #[derive(Debug, Copy, Clone, PartialEq, Eq)]
+    struct ExpressionContextFlags: u8 {
+        /// This flag is set when the `in` keyword should be included as part of a comparison
+        /// expression. It is to avoid ambiguity in `for ... in ...` statements. This flag is
+        /// set by default.
+        const INCLUDE_IN = 1 << 0;
+
+        /// This flag is set when a starred expression should be allowed. This doesn't affect the
+        /// parsing of a starred expression as it will be parsed nevertheless. But, if it is not
+        /// allowed, an error is reported.
+        const ALLOW_STARRED_EXPRESSION = 1 << 1;
+
+        /// This flag is set when the value of a starred expression should be limited to bitwise OR
+        /// precedence. Matches the `* bitwise_or` grammar rule if set.
+        const BITWISE_OR_PRECEDENCE = 1 << 2;
```

Comment:
Nit: The name here doesn't really fit with the other flags and it seems specific to `star` expression but the name doesn't say so. 

`STARRED_BITWISE_OR_PRECEDENCE`, `ALLOW_STARRED_CONDITIONAL_PRECEDENCE` or `DISALLOW_STARRED_CONDITIONAL_PRECEDENCE`